### PR TITLE
Surface namer exceptions in the delegator API.

### PR DESCRIPTION
**Pull request by olix0r**
When resolving a name, a namer may return an Activity.Failed with an exception.
Previously, this manifested as a 500 error.  Instead, these errors should be
mapped into the DelegatorTree response i.e. so that clients may illustrate
where these failures occurred.

### CODE REVIEW

**Comment by olix0r** [comment]
To test the UI changes, I ran:
```
:; ./sbt examples/marathon:run
```
and then went to `localhost:9990/delegator?router=http#%2Fhttp%2F1.1%2FGET%2Fdog`

File: linkerd/admin/src/test/scala/io/buoyant/linkerd/admin/names/WebDelegatorTest.scala [hunk 1]
```diff
@@ -40,15 +38,23 @@ class WebDelegatorTest extends FunSuite with Awaits {
...
+              |"dst":"/$/inet/127.1/8080"},
+            |"bound":{"addr":{"type":"bound","addrs":[{"ip":"127.0.0.1","port":8080}],"meta":{}},
+              |"id":"/$/inet/127.1/8080","path":"/"}}},
+          |{"type":"delegate","path":"/beh/humbug",
+            |"dentry":{"prefix":"/foo","dst":"/bah | /beh | /$/fail"},"delegate":{
+              |"type":"exception","path":"/error/humbug","dentry":{"prefix":"/beh","dst":"/error"},
+                |"message":"error naming /error/humbug"}},
+          |{"type":"fail","path":"/$/fail/humbug",
+            |"dentry":{"prefix":"/foo","dst":"/bah | /beh | /$/fail"}}]}}
+      |""".stripMargin.replaceAllLiterally("\n", ""))
```
Comment (esbie) [inline linkerd/admin/src/test/scala/io/buoyant/linkerd/admin/names/WebDelegatorTest.scala]: I have such a love/hate relationship with this test :joy:

File: linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/delegate.js [hunk 2]
```diff
@@ -63,9 +63,10 @@ function renderNode(obj){
   switch(obj.type) {
     case "delegate": obj.isDelegate = true; obj.child = renderNode(obj.delegate); break;
     case "alt": obj.isAlt = true; obj.child = obj.alt.map(function(e,i){ return renderNode(e); }).join(""); break;
-    case "neg": obj.isNeg = true; obj.break;
+    case "neg": obj.isNeg = true; break;
```
Comment (esbie) [inline linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/delegate.js]: nice catch!

**Comment by esbie** [comment]
:star: this is great and the tests are great #great

File: linkerd/admin/src/main/scala/io/buoyant/linkerd/admin/names/Delegator.scala [hunk 3]
```diff
@@ -26,10 +27,11 @@ object DelegateTree {
...
 
-  private def map[T, U](tree: DelegateTree[T], f: T => U): DelegateTree[U] = tree match {
-    case tree@Empty(_, _) => tree
-    case tree@Fail(_, _) => tree
-    case tree@Neg(_, _) => tree
+  private def map[T, U](orig: DelegateTree[T], f: T => U): DelegateTree[U] = orig match {
+    case tree: Exception => tree
+    case tree: Empty => tree
+    case tree: Fail => tree
+    case tree: Neg => tree
```
Comment (gtcampbell) [inline linkerd/admin/src/main/scala/io/buoyant/linkerd/admin/names/Delegator.scala]: tioli: If you like, you can shorten this to
```scala
case (_: Exception | _: Empty | _: Fail | _ : Neg) => orig
```

**Comment by gtcampbell** [comment]
:star: I agree, the tests are :100:
